app/controllers/catalog_controller.rb

```diff
@@ -96,8 +96,14 @@ def show
   end
 
   def parent_document
-    return unless params[:parent_id]
-    _, @parent_document = fetch(params[:parent_id])
+    if params[:parent_id]
+      _, @parent_document = fetch(params[:parent_id])
+    # we know in our data a fileset never has more than one parent; grab it for breadcrumb convenience
+    elsif fetch(params[:id]).first.documents.first[:internal_resource_ssim].include? "FileSet"
```

Comment:
idk; when I try to get `@document` it's nil.

Reply:
Looks like `parent_document` is being run as a `before_action`, but only for `show`.  Maybe if you move the `parent_document` call into `show`, you could do it after the solr document was loaded?

Reply:
I think because this method is called as a `before_action` the `@document` hasn't been loaded yet. `@document` is loaded here https://github.com/projectblacklight/blacklight/blob/master/app/controllers/concerns/blacklight/catalog.rb#L45